audit/audit_syslog_storage_helper.cc

```diff
@@ -93,15 +93,25 @@ future<> audit_syslog_storage_helper::write(const audit_info* audit_info,
     auto now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
     tm time;
     localtime_r(&now, &time);
-    sstring msg = seastar::format("<{}>{:%h %e %T} scylla-audit: \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\"",
+    std::string escaped_query;
+    escaped_query.reserve(audit_info->query().size() * 1.2);
+    for (auto c : audit_info->query()) {
+        if (c == '"') {
+            escaped_query.push_back('\\');
+            escaped_query.push_back('"');
+        } else {
+            escaped_query.push_back(c);
```

Comment:
This escaping is not 100% rigorous. Consider a query **ending** with backslash (this is not a real useful query, but a malicious client can still send it). It will cause you to output a backslash followed by doublequotes, and the reader will think this is an escaped quotes, not the end of the query.

If you use CSV's quoting technique (see https://datatracker.ietf.org/doc/html/rfc4180) you'll only double the doublequotes, and will not bring another character (backslash) into the party. However, if you backslashes (following the quoting technique of JSON, shell and C), you will need to add backslash in front of both double quotes and the backslash itself.

Reply:
I am escaping both `\` and `"` now, thanks, that's a good catch.